src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -447,7 +437,7 @@ where
     (
         export_collections,
         progress,
-        health.concat(&derived_health),
+        health.concatenate(health_streams),
```

Comment:
Correct me if I'm wrong, but I think this changes the progress tracking for the health output: previously, it was disconnected, but now the progress information from all sources flows into it. Looking through the code, we might not actually care, in the past the health cap was either 0 or dropped?

Reply:
The health stream is a bit of a weirdo and its progress events aren't used in anything. It communicates "writes that need to happen" which eventually the controllers write out to the system table at whatever timestamp it happens to be at. So I opted for the simpler API that doesn't need me to construct connection vectors